app/pages/sections/order-items/catalogue-solutions/select/routes.js

```diff
@@ -234,7 +234,13 @@ export const catalogueSolutionsSelectRoutes = (authProvider, addContext, session
       req, key: sessionKeys.selectedItemName,
     });
 
-    const { commencementDate } = await getCommencementDate({ orderId, accessToken });
+    let commencementDate = sessionManager.getFromSession({
+      req, key: sessionKeys.plannedDeliveryDate,
+    });
+
+    if (!commencementDate) {
+      ({ commencementDate } = await getCommencementDate({ orderId, accessToken }));
+    }
```

Comment:
Can you make use of `getFromSessionOrApi` here? I think it might fit. There are some examples in `app/helpers/routes` – it might be worth creating a new helper for it.

Reply:
yeah good shout will change this now